scripts/eth-allowed-packages.txt

```diff
@@ -25,6 +25,8 @@
 "github.com/ava-labs/libevm/ethdb/pebble"
 "github.com/ava-labs/libevm/event"
 "github.com/ava-labs/libevm/libevm"
+"github.com/ava-labs/libevm/libevm/legacy"
+"github.com/ava-labs/libevm/libevm/pseudo"
```

Comment:
Why is there a need for `pseudo` access? This is concerning and suggests I did something wrong in `libevm`.

Reply:
It's used in a `TestCopyHeader` test with reflection, I removed the import by replacing it with a reflect type string check. I may have been a bit far on the testing of copy header but it did help me debunk some problems 1-2 weeks ago.

Reply:
I think that's perfectly valid and it's absolutely cleaner to do that instead of using a reflect string. In case you're unaware, if a type implements https://pkg.go.dev/github.com/ava-labs/libevm/libevm/pseudo#EqualityChecker then you can use `pseudo.Type.Equal()` and if you're using `cmp.Diff()` then you can use:

```go
opt := cmp.Comparer((*pseudo.Type).Equal)
if diff := cmp.Diff(want, got, opt); diff != "" {
```

If you're testing, for example `HeaderExtra` then you can even implement `func (*HeaderExtra) Equal(*HeaderExtra) bool` in the _test_ file so it doesn't pollute the `types` package.

I'll implement `package pseudotest` to make this easier and so we can swap it out in the list of allowed packages.